identity-iota/src/did/doc/iota_document.rs

```diff
@@ -291,15 +278,19 @@ impl IotaDocument {
     self.document.also_known_as()
   }
 
-  /// Returns the default authentication method of the DID document.
-  pub fn authentication(&self) -> &IotaVerificationMethod {
-    // This `unwrap` is "fine" - a valid document will
-    // always have a resolvable authentication method.
-    let method: &MethodRef = self.document.authentication().head().unwrap();
-    let method: &VerificationMethod = self.document.resolve_method_ref(method).unwrap();
-
-    // SAFETY: We don't allow invalid authentication methods.
-    unsafe { IotaVerificationMethod::new_unchecked_ref(method) }
+  /// Returns the first [`IotaVerificationMethod`] with a capability invocation relationship
+  /// capable of signing this DID document.
+  pub fn default_signing_method(&self) -> Result<&IotaVerificationMethod> {
```

Comment:
Or if the developer removes all signing keys (even if it's temporary while rotating them for example), since `remove_method` no longer disallows removing signing keys at all.